traffic_ops/traffic_ops_golang/api/generic_crud.go

```diff
@@ -118,28 +125,86 @@ func GenericCreateNameBasedID(val GenericCreator) (error, error, int) {
 	return nil, nil, http.StatusOK
 }
 
-func GenericRead(val GenericReader) ([]interface{}, error, error, int) {
+func MakeFirstQuery(val GenericReader, h map[string][]string, where string, orderBy string, pagination string, queryValues map[string]interface{}) (bool, time.Time) {
+	var max time.Time
+	ims := []string{}
+	runSecond := true
+	if h == nil {
+		return runSecond, max
+	}
+	ims = h[rfc.IfModifiedSince]
+	if len(ims) == 0 {
+		return runSecond, max
+	}
+	if l, ok := web.ParseHTTPDate(ims[0]); !ok {
+		log.Warnf("Date not parsable %v", ims[0])
+		return runSecond, max
+	} else {
+		query := val.SelectMaxLastUpdatedQuery(where, orderBy, pagination, val.GetType())
+		rows, err := val.APIInfo().Tx.NamedQuery(query, queryValues)
+		defer rows.Close()
+		if err != nil {
+			log.Warnf("Couldn't get the max last updated time: %v", err)
+			return runSecond, max
+		}
+		if err == sql.ErrNoRows {
+			runSecond = false
```

Comment:
instead of setting `runSecond = false` before returning it, would it be more clear to create a 2nd variable at the top - `dontRunSecond = false` - and return that instead?

Reply:
I was just trying to minimize the number of variables that I use in the function. I always find it easier to read a piece of code if there is one variable tracking the state of the system, rather than multiple. Just a personal preference, but let me know if you'd like me to change it.

Reply:
I find that unnecessary state changes make the code harder to follow -- since it is primarily just the return value, there's no real need for it to change state -- they could be `const` too.

`return dontRunSecond, max` and `return runSecond, max` makes it more clear IMO, but if you don't want to change it that's fine too. Just a suggestion.

Reply:
I changed it.